### ISSUE TITLE
TabbedUI breaks content fields with `position: 'sidebar'`

### ISSUE THREAD

**Issue by omavi** [payloadcms/plugin-seo#35]
The following logic doesn't account for fields that should end up in the sidebar: https://github.com/payloadcms/plugin-seo/blob/348b519b9178cf3915bddb1a7d1b15863051693c/src/index.ts#L82-L98

**Comment by jacobsfletch** [comment]
@omavi are you saying you have `position: 'sidebar'` set on your base fields, and those are rendering in the "content" tab instead of the sidebar?

**Comment by omavi** [comment]
@jacobsfletch yes, exactly!

**Comment by qbeauperin** [comment]
Can confirm, just stumbled upon this issue as well.

**Comment by qbeauperin** [comment]
Actually, after looking a bit more into this, I think there's two issues at play here:

1. A field's `admin.position` can only be defined if that field is at the first level of the collection's `fields` array (eg: a field in a `Tabs` or a `Group` field will currently not be rendered in the sidebar)
2. If your first field is a `Tabs` field, this plugin will discard any other fields defined after this first `Tabs` field when generating the SEO tab.

**Comment by ssyberg** [comment]
Hmm I tried explicitly defining my fields in a "Content" tab and pulling the sidebar fields out of the tabs completely but that resulted in some even stranger behavior:
![image](https://user-images.githubusercontent.com/1661046/228030956-a1194df1-dd32-48f3-9928-1bf8c5c94afb.png)

**Comment by qbeauperin** [comment]
The workaround I've ended up using is not using the plugin, but using the plugin's fields directly.

``` typescript
import { CollectionConfig } from 'payload/types';

// SEO fields
import { Overview } from '@payloadcms/plugin-seo/dist/ui/Overview';
import { getMetaDescriptionField } from '@payloadcms/plugin-seo/dist/fields/MetaDescription';
import { getMetaTitleField } from '@payloadcms/plugin-seo/dist/fields/MetaTitle';
import { getPreviewField } from '@payloadcms/plugin-seo/dist/ui/Preview';
import { getMetaImageField } from '@payloadcms/plugin-seo/dist/fields/MetaImage';

const seoConfig = {
  uploadsCollection: 'media',
  generateTitle: ({ doc }) => `${doc?.title?.value} — Site name`,
  generateDescription: ({ doc }) => doc?.excerpt?.value,
  generateURL: ({ doc }) => `https://faefarm.com/news/${doc?.slug?.value}`,
  generateImage: ({ doc }) => doc?.featuredImage?.value,
}

const Posts: CollectionConfig = {
  slug: 'posts',
  fields: [
    {
      type: 'tabs',
      tabs: [
        {
          label: 'Content',
          fields: [
            {
              name: 'title',
              type: 'text',
              localized: true,
              required: true,
            },
            {
              name: 'slug',
              type: 'text',
              localized: true,
              required: true,
              unique: true,
            },
            {
              name: 'content',
              type: 'richText',
              localized: true,
            },
          ]
        },
        {
          label: 'SEO',
          fields: [
            {
              name: 'meta',
              label: 'SEO',
              type: 'group',
              fields: [
                {
                  name: 'overview',
                  label: 'Overview',
                  type: 'ui',
                  admin: {
                    components: {
                      Field: Overview,
                    },
                  },
                },
                {
                  name: 'title',
                  type: 'text',
                  localized: true,
                  admin: {
                    components: {
                      Field: (props) => getMetaTitleField({ ...props, ...{ pluginConfig: seoConfig } }),
                    },
                  },
                },
                {
                  name: 'description',
                  type: 'textarea',
                  localized: true,
                  admin: {
                    components: {
                      Field: (props) => getMetaDescriptionField({ ...props, ...{ pluginConfig: seoConfig } }),
                    },
                  },
                },
                {
                  name: 'image',
                  label: 'Meta Image',
                  type: 'upload',
                  localized: true,
                  relationTo: seoConfig?.uploadsCollection,
                  admin: {
                    description: 'Maximum upload file size: 12MB. Recommended file size for images is <500KB.',
                    components: {
                      Field: (props) => getMetaImageField({ ...props, ...{ pluginConfig: seoConfig } }),
                    },
                  },
                },
                {
                  name: 'preview',
                  label: 'Preview',
                  type: 'ui',
                  admin: {
                    components: {
                      Field: (props) => getPreviewField({ ...props, ...{ pluginConfig: seoConfig } }),
                    },
                  },
                },
              ]
            }
          ]
        }
      ]
    },
    {
      name: 'publishedDate',
      type: 'date',
      admin: {
        position: 'sidebar',
      },
    },
  ],
}

export default Posts;
```

**Comment by ssyberg** [comment]
I implemented a workaround in a separate plugin that moves all `sidebar` fields to the top of `CollectionConfig.fields`

**Comment by jacobsfletch** [comment]
> Actually, after looking a bit more into this, I think there's two issues at play here:
> 
> 1. A field's `admin.position` can only be defined if that field is at the first level of the collection's `fields` array (eg: a field in a `Tabs` or a `Group` field will currently not be rendered in the sidebar)
> 2. If your first field is a `Tabs` field, this plugin will discard any other fields defined after this first `Tabs` field when generating the SEO tab.

@qbeauperin you are spot on here, I just opened PR #47 with the fix. I've also updated the README and any relevant code with notes explaining exactly what `tabbedUI` does to your config and how to render fields outside the tabs. Here's the gist:

If your collection is not already tab-enabled, meaning the first field in your config is not of type `tabs`, then one will be created for you called `Content`. If you wish to continue to use top-level or sidebar fields with `tabbedUI`, you  must not let the default `Content` tab get created for you. Instead, you must define the first field of your config with type `tabs` and place all other fields adjacent to this one at the top-level. There's an example of that [here](https://github.com/payloadcms/plugin-seo/blob/main/demo/src/collections/Posts.ts).

THANK YOU everyone for your patience on this. It looks like we've had a handle of others waiting on this fix, too. I'll report back as soon as possible, as soon as a release has been published.
